compiler/rustc_const_eval/src/const_eval/eval_queries.rs

```diff
@@ -226,16 +226,7 @@ pub fn eval_to_const_value_raw_provider<'tcx>(
     key: ty::ParamEnvAnd<'tcx, GlobalId<'tcx>>,
 ) -> ::rustc_middle::mir::interpret::EvalToConstValueResult<'tcx> {
     // see comment in eval_to_allocation_raw_provider for what we're doing here
```

Comment:
That comment now probably needs updating?

The assertion error or the comment should explain what a rustc dev should do when they hit this panic.